apps/aevm/src/aevm_eeevm_state.erl

```diff
@@ -84,6 +84,11 @@ init(#{ env  := Env
 
     ChainState = maps:get(chainState, Env),
     ChainAPI =  maps:get(chainAPI, Env),
+    {IsOffChain, OnChainTrees} =
+        case maps:get(off_chain, Env) of
```

Comment:
Based on code inspection, this function is always called with this `false`. Can this field be removed altogether from this representation? (In the code, the this env that is passed to this function is always created from scratch - not passed over from arguments - so it shall be same to remove this field altogether.) The code is a bit complex to follow, and removing this field from this point would ease that.